SoftFox/SoftFox/SoftFox.cpp

```diff
@@ -112,12 +112,47 @@ void SoftFox::run()
 			playerX -= PLAYER_MOVEMENT_SPEED;
 		if (keyboardState[SDL_SCANCODE_RIGHT])
 			playerX += PLAYER_MOVEMENT_SPEED;
+		
+
+		SDL_Rect playerBox = { playerX, playerY, tileSize / 2, tileSize };
+
+		playerY += gravity;
+		
 		/*
-		if (physics->isCollision(playerSprite, platformSprite))
+		int platformX = tileSize * level->getWallX() + tileSize / 2;// + tileSize;
+		int platformY = tileSize * level->getWallY() + tileSize / 2;// - tileSize/4;
+		SDL_Rect platformBox = { platformX, platformY, tileSize, tileSize };
+
+		if (physics->isCollision(playerBox, platformBox))
 		{
-			PLAYER_MOVEMENT_SPEED = 0;
+			playerY -= upForce;
 		}
 		*/
+		
+		for (int y = 0; y < level->getHeight(); y++)
+		{
+			for (int x = 0; x < level->getWidth(); x++)
+			{
+				if (level->isWall(x, y))
+				{
+					int platformX[] = { tileSize * level->getWallX() + tileSize / 2 };
+					int platformY[] = { tileSize * level->getWallY() + tileSize / 2 };
+					for (int a = 0; a < size_t (platformX); a++)
```

Comment:
`size_t(platformX)` reinterprets the pointer `platformX` as a number -- almost certainly not what you wanted. Maybe you meant `sizeof(platformX)` which would give the size of the array `platformX` in bytes -- although again I don't think that's what you want.